.github/workflows/spike-openocd-tests.yml

```diff
@@ -125,7 +126,30 @@ jobs:
               --gcc /opt/riscv/toolchain/bin/riscv-none-elf-gcc \
               --gdb /opt/riscv/toolchain/bin/riscv-none-elf-gdb \
               --sim_cmd /opt/riscv/spike/bin/spike \
-              --server_cmd $GITHUB_WORKSPACE/src/openocd
+              --remotelogfile-enable \
+              --server_cmd "$GITHUB_WORKSPACE/src/openocd -d3"
+
+      - name: Run Spike64-2-hwthread Tests
+        if: success() || steps.spike32-tests.conclusion == 'failure'
+        run: |
+          cd riscv-tests/debug
+          ./gdbserver.py targets/RISC-V/spike64-2-hwthread.py --print-failures \
+              --gcc /opt/riscv/toolchain/bin/riscv-none-elf-gcc \
+              --gdb /opt/riscv/toolchain/bin/riscv-none-elf-gdb \
+              --sim_cmd /opt/riscv/spike/bin/spike \
+              --remotelogfile-enable \
+              --server_cmd "$GITHUB_WORKSPACE/src/openocd -d3"
+
+      - name: Run Spike32-2-hwthread Tests
+        if: success() || steps.spike32-tests.conclusion == 'failure'
+        run: |
+          cd riscv-tests/debug
+          ./gdbserver.py targets/RISC-V/spike32-2-hwthread.py --print-failures \
+              --gcc /opt/riscv/toolchain/bin/riscv-none-elf-gcc \
+              --gdb /opt/riscv/toolchain/bin/riscv-none-elf-gdb \
+              --sim_cmd /opt/riscv/spike/bin/spike \
+              --remotelogfile-enable \
+              --server_cmd "$GITHUB_WORKSPACE/src/openocd -d3"
```

Comment:
In past, I have observed that some of the riscv-tests were unstable.

For that reason, I'd recommend to run these newly added tests many times (e.g. 50 times on your local machine) to detect any potential instability, or to gain high level of confidence that the tests are very stable.

Reply:
> In past, I have observed that some of the riscv-tests were unstable.
> (Maybe you have already done that.)

Well actually we've already done that. The tests are quite stable now (at least on a decent machines). Internally in Syntacore I've analysed the stability of the teststuite and run the whole testsuite more than a 1000 times. Not a single failure was detected. That being said there is still this https://github.com/riscv-collab/riscv-openocd/issues/104 that still may haunt us under certain conditions - I know the root cause, but still could not find time to fix this.

Reply:
Thank you.